### PR TITLE
[python][docs] add info on adaptive learning rate in the sklearn API

### PR DESCRIPTION
**Pull request by mlisovyi**
A small add-on to the docs to clarify how users of the sklearn API can implement shrinkage of the learning rate via a callback
The PR follows the discussion in #1317

### CODE REVIEW

**Comment by StrikerRUS** [comment]
@mlisovyi Thanks for the PR!

I suppose, this description is too long and isn't related to the describing parameter directly. Common users don't want to know such detailed example. I think, something like "You can use ``callbacks`` parameter of ``fit`` method to schedule learning rate by ``reset_parameter`` callback" will be enough. Common user will read this and go to callbacks description to learn more, while experienced user will just recollect described feature in his memory.

**Comment by mlisovyi** [comment]
@StrikerRUS OK, that makes sense. I'll reimplement it later today following your proposal

**Comment by mlisovyi** [comment]
I've shortened the learning_rate description but kept a note that this would ignore the actual argument value. Let me know if you do not like it.

I've also made another small update to the docs following #1355

**Comment by StrikerRUS** [comment]
@mlisovyi Thanks! LGTM!
But what about to add a phrase ".. to shrink/adapt learning rate **while training** ..."?

**Comment by mlisovyi** [comment]
This indeed makes the statement more clear, but i changed it to "*in* training" (following my poor English taste :smile:  )

**Comment by StrikerRUS** [comment]
Sure! 😃 

Now please fix whitespaces and I'll be able to merge.
image

**Comment by mlisovyi** [comment]
fixed, apologies, I overlooked that

**Comment by StrikerRUS** [comment]
Thank you for your contribution!

**Comment by StrikerRUS** [comment]
@guolinke I cannot merge due to broken GPU tests. If you have no comments, please merge.
